homeassistant/components/binary_sensor/insteon_plm.py

```diff
@@ -15,38 +15,43 @@
 
 _LOGGER = logging.getLogger(__name__)
 
+SENSOR_TYPES = {'openClosedSensor': 'opening',
+                'motionSensor': 'motion',
+                'doorSensor': 'door',
+                'leakSensor': 'moisture'}
+
 
 @asyncio.coroutine
 def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
     """Set up the INSTEON PLM device class for the hass platform."""
+    state_list = []
     plm = hass.data['insteon_plm']
 
-    device_list = []
-    for device in discovery_info:
-        name = device.get('address')
-        address = device.get('address_hex')
-
-        _LOGGER.info('Registered %s with binary_sensor platform.', name)
+    for device_info in discovery_info:
+        address = device_info['address']
+        device = plm.devices[address]
+        state_key = device_info['state_key']
 
-        device_list.append(
-            InsteonPLMBinarySensorDevice(hass, plm, address, name)
-        )
+        state_list.append(InsteonPLMBinarySensor(hass,
+                                                 device,
+                                                 state_key))
 
-    async_add_devices(device_list)
+    async_add_devices(state_list)
 
 
-class InsteonPLMBinarySensorDevice(BinarySensorDevice):
-    """A Class for an Insteon device."""
+class InsteonPLMBinarySensor(BinarySensorDevice):
+    """A Class for an Insteon device state."""
 
-    def __init__(self, hass, plm, address, name):
-        """Initialize the binarysensor."""
+    def __init__(self, hass, device, state_key):
+        """Initialize the INSTEON PLM binary sensor."""
         self._hass = hass
-        self._plm = plm.protocol
-        self._address = address
-        self._name = name
+        self._insteon_device_state = device.states[state_key]
+        self._insteon_device = device
+        self._sensor_type = SENSOR_TYPES.get(self._insteon_device_state.name,
+                                             None)
```

Comment:
`None` is the default value returned if the key is missing.

Reply:
Yes, this is by design. If you look at the `BinarySensorDevice` base class, the default value of the `device_class` property is `None`. So if the underlying library produces a binary sensor that is not in one of the `SENSOR_TYPES` it should default to `None` and produce the default behavior of `BinarySensorDevice`.

Reply:
Yeah, I just would like you to remove the explicit default of `None` here, cause it's already the default default for `dict.get`.